### PR TITLE
[BEAM-1379] Shade Guava in beam-sdks-java-io-kafka module

### PR DESCRIPTION
**Pull request by aviemzur**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

### CODE REVIEW

File: sdks/java/io/kafka/pom.xml [hunk 1]
```diff
@@ -40,6 +40,35 @@
...
+                <artifactSet>
+                  <includes>
+                    <include>com.google.guava:guava</include>
+                  </includes>
+                </artifactSet>
+                <relocations>
+                  <relocation>
+                    <pattern>com.google.common</pattern>
+                    <shadedPattern>org.apache.beam.spark.relocated.com.google.common</shadedPattern>
+                  </relocation>
```
Comment (dhalperi) [inline sdks/java/io/kafka/pom.xml]: use `org.apache.beam.io.kafka.relocated`?

Comment (davorbonaci) [inline sdks/java/io/kafka/pom.xml]: also `relocated` -> `repackaged`, to follow the pattern previously established? (or change in both places)

File: sdks/java/io/kafka/pom.xml [hunk 2]
```diff
@@ -40,6 +40,35 @@
...
+              <goals>
+                <goal>shade</goal>
+              </goals>
+              <configuration>
+                <artifactSet>
+                  <includes>
+                    <include>com.google.guava:guava</include>
+                  </includes>
+                </artifactSet>
+                <relocations>
```
Comment (davorbonaci) [inline sdks/java/io/kafka/pom.xml]: Random comment, nothing to do now:

I worry that in a future version a shaded dependency will fall out of the relocations specified here, and we won't know. If there's an easy ability to "assert" that the final jar contains nothing outside `org.apache.beam.<component>` would be awesome.

**Comment by dhalperi** [comment]
R: @dhalperi @davorbonaci

File: sdks/java/io/kafka/pom.xml [hunk 3]
```diff
@@ -40,6 +40,35 @@
...
+                  <relocation>
+                    <pattern>com.google.common</pattern>
+                    <shadedPattern>org.apache.beam.spark.relocated.com.google.common</shadedPattern>
```
Comment (davorbonaci) [inline sdks/java/io/kafka/pom.xml]: not spark right?

File: sdks/java/io/kafka/pom.xml (hunk 1 above)
Comment (aviemzur) [inline sdks/java/io/kafka/pom.xml]: @dhalperi Yup, copy paste mistake.

@davorbonaci I'll change in spark runner pom as well then

File: sdks/java/io/kafka/pom.xml (hunk 2 above)
Comment (aviemzur) [inline sdks/java/io/kafka/pom.xml]: @davorbonaci 

IIUC you mean you want to make sure that every class that was shaded into the jar (And every class packaged into the jar) begins with `org.apache.beam.`?

**Comment by aviemzur** [comment]
@dhalperi @davorbonaci made the renames you asked for.
CC: @amitsela

**Comment by aviemzur** [comment]
Retest this please.

File: runners/spark/pom.xml [hunk 4]
```diff
@@ -383,11 +383,11 @@
                     version used by Hadoop (v11) -->
                   <relocation>
                     <pattern>com.google.common</pattern>
-                    <shadedPattern>org.apache.beam.spark.relocated.com.google.common</shadedPattern>
```
Comment (amitsela) [inline runners/spark/pom.xml]: 👍

**Comment by dhalperi** [comment]
LGTM, thanks @aviemzur !

**Comment by dhalperi** [comment]
(and merged)

File: sdks/java/io/kafka/pom.xml (hunk 2 above)
Comment (davorbonaci) [inline sdks/java/io/kafka/pom.xml]: I probably wouldn't go as far as a new plugin; something easy/simple only.

**Comment by davorbonaci** [comment]
Thanks @aviemzur!

**Pull request closed without merging**